src/libs/actions/Report.ts

```diff
@@ -1607,6 +1609,59 @@ function readNewestAction(reportID: string | undefined, shouldResetUnreadMarker
     }
 }
 
+function markAllMessagesAsRead() {
+    const lastReadTime = DateUtils.getDBTimeWithSkew();
+
+    const optimisticUnreadReports: Record<string, Pick<Report, 'lastReadTime'>> = {};
+    const failureUnreadReports: Record<string, Pick<Report, 'lastReadTime'>> = {};
+    const reportIDs: string[] = [];
+    Object.values(allReports ?? {}).forEach((report) => {
+        if (!report) {
+            return;
+        }
+
+        const oneTransactionThreadReportID = ReportActionsUtils.getOneTransactionThreadReportID(
+            report.reportID,
+            allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`],
+        );
+        const oneTransactionThreadReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${oneTransactionThreadReportID}`];
+        if (!isUnread(report, oneTransactionThreadReport)) {
+            return;
+        }
+
+        const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`;
+        optimisticUnreadReports[reportKey] = {lastReadTime};
+        failureUnreadReports[reportKey] = {lastReadTime: report.lastReadTime};
+        reportIDs.push(report.reportID);
+    });
+
+    if (reportIDs.length === 0) {
+        return;
+    }
+
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE_COLLECTION,
+            key: ONYXKEYS.COLLECTION.REPORT,
+            value: optimisticUnreadReports,
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE_COLLECTION,
+            key: ONYXKEYS.COLLECTION.REPORT,
+            value: failureUnreadReports,
+        },
+    ];
+
+    const parameters: MarkAllMessagesAsReadParams = {
+        reportIDs,
```

Comment:
Can we please update this to be be `reportIDList`. That is what the BE expects. Otherwise, I tested it locally with my back-end and things seem to work well.